src/main/java/net/dries007/tfc/client/render/blockentity/ChannelBlockEntityRenderer.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the EUPL, Version 1.2.
+ * You may obtain a copy of the Licence at:
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+package net.dries007.tfc.client.render.blockentity;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.mojang.blaze3d.vertex.VertexConsumer;
+
+import net.dries007.tfc.client.RenderHelpers;
+import net.dries007.tfc.common.blockentities.ChannelBlockEntity;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.MultiBufferSource;
+import net.minecraft.client.renderer.RenderType;
+import net.minecraft.client.renderer.blockentity.BlockEntityRenderer;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.core.Direction;
+import net.minecraft.core.registries.BuiltInRegistries;
+import net.minecraft.world.level.material.Fluid;
+import net.minecraft.world.phys.AABB;
+import net.minecraft.world.phys.Vec3;
+import net.neoforged.neoforge.client.extensions.common.IClientFluidTypeExtensions;
+
+public class ChannelBlockEntityRenderer implements BlockEntityRenderer<ChannelBlockEntity> {
+    @Override
+    public void render(ChannelBlockEntity channel, float partialTicks, PoseStack poseStack, MultiBufferSource buffer,
+            int combinedLight, int combinedOverlay) {
+        if (!channel.hasFlow()) {
+            return;
+        }
+
+        Fluid fluid = BuiltInRegistries.FLUID.get(channel.getFluid());
```

Comment:
we prefer showing `final` on functionally final local variables, but not necessarily required